### ISSUE TITLE
[feature request] Add retry failed trial callback to optuna study

### ISSUE THREAD

**Issue by isaacncz** [DLR-RM/rl-baselines3-zoo#263]
I was training on custom environment and some trials failed in the middle of the study. 
I notice optuna offers RetryFailedTrialCallback [https://optuna.readthedocs.io/en/latest/reference/generated/optuna.storages.RetryFailedTrialCallback.html](url).

would like to suggest adding this to the cli command too.

**Comment by araffin** [comment]
Hello,

> would like to suggest adding this to the cli command too.

how would you like it to be done?
by passing an argument `--retry-failed-trial`?
and what is your use case?

Usually a failed trial in RL is because of inf/NaN in the env or inf/NaN in the model (due to high learning rate for instance), so retrying one will usually lead to a failure again.

**Comment by isaacncz** [comment]
I developed the environment which requires two computers (1 running on Windows, 1 on Ubuntu) for training. Sometimes the disconnection happens and the current hyperparameter will be skipped when restarting the training. Therefore I suggest to add argument for user to specify the number of retries. By default can be 0

**Comment by araffin** [comment]
> I suggest to add argument for user to specify the number of retries

sounds good, please go ahead with a PR ;) (don't forget to take a look at the PR checklist)

**Comment by isaacncz** [comment]
@araffin didn't notice it's an experimental feature, will it be ok for PR?

**Comment by araffin** [comment]
Let's wait for it to be non experimental.
In the meantime, you can always use that feature if you have a fork of the RL Zoo ;)
